### PR TITLE
Fix #2002

### LINKED ISSUE #2002: Table heading not resizing properly
I am seeing an issue where the table heading row isn't responding correctly to the window resizing.  If you load the table when the browser is not full screen and resize the browser to full screen the heading row isn't lining up the table content.  I can reproduce the error using the examples site as well, see the attached screen shot.  The reverse is a problem as well (from full screen to smaller).

image

### PR DESCRIPTION
**Pull request by djhvscf**
Added a built-in resize event where we fit the header automatically when
the user resizes the window

### CODE REVIEW

File: src/bootstrap-table.js [hunk 1]
```diff
@@ -2168,6 +2178,7 @@
             this.$tableHeader.show();
             this.resetHeader();
             padding += this.$header.outerHeight();
+            fitHeaderOnDocumentResizeEvent(this);
```
Comment (wenzhixin) [inline src/bootstrap-table.js]: I think here `fitHeaderOnDocumentResizeEvent ` maybe be called many times.

Comment (djhvscf) [inline src/bootstrap-table.js]: Jummm yes.. you're right... any advice?

**Comment by djhvscf** [comment]
Example with the issue:
http://jsfiddle.net/djhvscf/9ymkp3tm/6/
Change the view of the table, when we have the card view we can see that the footer is shown.

Example with the issue fixed:
http://jsfiddle.net/djhvscf/9ymkp3tm/7/

**Comment by wenzhixin** [comment]
@djhvscf  Like this:
```js
        $(window).off('resize', $.proxy(this.fitHeader, this));
        if (!this.options.showHeader || this.options.cardView) {
            this.$header.hide();
            this.$tableHeader.hide();
            this.$tableLoading.css('top', 0);
        } else {
            this.$header.show();
            this.$tableHeader.show();
            this.$tableLoading.css('top', this.$header.outerHeight() + 1);
            // Assign the correct sortable arrow
            this.getCaret();
            $(window).on('resize', $.proxy(this.fitHeader, this));
        }
```

**Comment by wenzhixin** [comment]
Help to squash the commits together after update the code.

https://robots.thoughtbot.com/git-interactive-rebase-squash-amend-rewriting-history. for example `git rebase -i HEAD~9`:

```
# Rebase 8db7e8b..fa20af3 onto 8db7e8b
#
# Commands:
#  p, pick = use commit
#  r, reword = use commit, but edit the commit message
#  e, edit = use commit, but stop for amending
#  s, squash = use commit, but meld into previous commit
#  f, fixup = like "squash", but discard this commit's log message
#  x, exec = run command (the rest of the line) using shell
#
# These lines can be re-ordered; they are executed from top to bottom.
#
# If you remove a line here THAT COMMIT WILL BE LOST.
#
# However, if you remove everything, the rebase will be aborted.
#
# Note that empty commits are commented out
```

**Comment by djhvscf** [comment]
Example: 

using the resize method: http://jsfiddle.net/9ymkp3tm/8/
not using the resize method: http://jsfiddle.net/e3nk137y/5037/

@wenzhixin

**Comment by wenzhixin** [comment]
:+1:

**Pull request merged**